04.ls/ls.rb

```diff
@@ -1,9 +1,12 @@
 # frozen_string_literal: true
 
+require 'optparse'
+
 COLUMN_SIZE = 3
 
 def main
-  files = Dir.children(ARGV[0] || '.').reject { |file| file[0] == '.' }.sort
+  option_judgment = ARGV.getopts('a')['a'] ? ['\.*', '*'] : '*'
```

Comment:
```suggestion
  option_judgment = ['*']
  option_judgment << '.*' if ARGV.getopts('a')['a']
```

`*`は共通なので、上のようにした方が意図は明確になるかもしれません。
あと、`.*`はバックスラッシュ無しで同じように動くはずです〜。

それから `option_judgment` という変数名ですが `judgment` というのはあまりプログラミングでは見かけない単語ですね。
`Dir.glob`に渡す第1引数なので、こういうときはAPIドキュメントに書かれている名前に合わせると良いかもしれません。

https://docs.ruby-lang.org/ja/latest/method/Dir/s/=5b=5d.html

Reply:
ご確認ありがとうございます🙇‍♂️

```ruby
option_judgment << '.*' if ARGV.getopts('a')['a']
```
このコードなのですが、
このコードだとaオプションが指定されているとき`*` → `.*`のような順番になってしまい、
自分のOSだと 、lsコマンドと挙動が変わってしまうのですがいかがでしょうか？（`.*` → `*`　この順番にしたいのです）
`unshift`メソッドを使ってはいけないでしょうか？
```ruby
files = Dir.glob(['*', '.*'], base: ARGV[0] || '.')
```